src/periphery/abstracts/SablierMerkleBase.sol

```diff
@@ -31,12 +31,18 @@ abstract contract SablierMerkleBase is
     /// @inheritdoc ISablierMerkleBase
     uint40 public immutable override EXPIRATION;
 
+    /// @inheritdoc ISablierMerkleBase
+    address public immutable FACTORY;
+
     /// @inheritdoc ISablierMerkleBase
     bytes32 public immutable override MERKLE_ROOT;
 
     /// @dev The name of the campaign stored as bytes32.
     bytes32 internal immutable NAME;
 
+    /// @inheritdoc ISablierMerkleBase
+    uint256 public immutable SABLIER_FEE;
```

Comment:
why don't we also remove the variable from the merkle contract and keep it only in factory?

```solidity
function claim(uint256 index, address recipient, uint128 amount, bytes32[] calldata merkleProof) external payable {
        // Check: `msg.value` is not less than the sablier fee.
        if (msg.value < FACTORY. SABLIER_FEE()) {
            revert Errors.SablierMerkleBase_InsufficientFeePayment(msg.value, SABLIER_FEE);
        }

        // --snip--
}
```

this approach might more expensive, as there is one more external call, but i believe this to be better as we have the flexibility to active all merkle campaigns, even after deployment

wdyt?

Reply:
> why don't we also remove the variable from the merkle contract and keep it only in factory?

Same reason as [given here](https://github.com/sablier-labs/v2-core/issues/1032#issuecomment-2337913704), so that Sablier admin cannot change the fee for active campaigns. It would be a bad idea to display one fee at the time of launching a campaign, and then update it later. Campaign creators may not like that.